clients/python/cacclient/client.py

```diff
@@ -0,0 +1,101 @@
+
+file_name = (
+    "libcac_client.dylib" if platform == "darwin"
+    else "libcac_client.dll" if platform == "linux"
+    else "libcac_client.so"
+)
+
+lib_path = os.path.join(lib_path, file_name)
+
+class CacClient:
+    rust_lib = ctypes.CDLL(lib_path)
+
+    rust_lib.cac_new_client.argtypes = [ctypes.c_char_p, ctypes.c_int, ctypes.c_char_p]
+    rust_lib.cac_new_client.restype = ctypes.c_int
+
+    rust_lib.cac_get_client.argtypes = [ctypes.c_char_p]
+    rust_lib.cac_get_client.restype = ctypes.c_char_p
+
+    rust_lib.cac_start_polling_update.argtypes = [ctypes.c_char_p]
+    rust_lib.cac_start_polling_update.restype = None
+
+    rust_lib.cac_free_client.argtypes = [ctypes.c_char_p]
+    rust_lib.cac_free_client.restype = None
+
+    rust_lib.cac_last_error_message.restype = ctypes.c_char_p
+
+    rust_lib.cac_get_config.argtypes = [ctypes.c_char_p, ctypes.c_char_p, ctypes.c_char_p]
+    rust_lib.cac_get_config.restype = ctypes.c_char_p
+
+    rust_lib.cac_last_error_length.restype = ctypes.c_int
+
+    rust_lib.cac_free_string.argtypes = [ctypes.c_char_p]
+    rust_lib.cac_free_string.restype = None
+
+    rust_lib.cac_get_last_modified.argtypes = [ctypes.c_char_p]
+    rust_lib.cac_get_last_modified.restype = ctypes.c_char_p
+
+    rust_lib.cac_get_resolved_config.argtypes = [ctypes.c_char_p, ctypes.c_char_p, ctypes.c_char_p, ctypes.c_char_p]
+    rust_lib.cac_get_resolved_config.restype = ctypes.c_char_p
+
+    rust_lib.cac_get_default_config.argtypes = [ctypes.c_char_p, ctypes.c_char_p]
+    rust_lib.cac_get_default_config.restype = ctypes.c_char_p
+
+    def __init__(self, tenant_name: str, polling_frequency: int, cac_host_name: str):
+        if not tenant_name or not cac_host_name:
+            raise ValueError("tenantName cannot be null or empty")
+
+        self.tenant = tenant_name
+        self.polling_frequency = polling_frequency
+        self.cac_host_name = cac_host_name
+
+    def get_cac_last_error_message(self) -> str:
+        return self.rust_lib.cac_last_error_message().decode()
+
+    def get_cac_last_error_length(self) -> int:
+        return self.rust_lib.cac_last_error_length()
+
+    def get_cac_client(self) -> str:
+        return self.rust_lib.cac_get_client(self.tenant.encode())
+
+    def create_new_cac_client(self) -> int:
+        resp = self.rust_lib.cac_new_client(
+            self.tenant.encode(), self.polling_frequency, self.cac_host_name.encode())
+        if resp == 1:
+            error_message = self.get_cac_last_error_message()
+            print("Some Error Occur while creating new client ", error_message)
+        return resp
+
+    def start_cac_polling_update(self):
+        threading.Thread(target=self._polling_update_worker).start()
+
+    def _polling_update_worker(self):
+        self.rust_lib.cac_start_polling_update(self.tenant.encode())
+
+    def get_cac_config(self, filter_query: str = "{}", filter_prefix: str = "") -> str:
```

Comment:
@namitgoel can we add filter_prefix: str = None, default as None instead of empty string in all arguments